EIPS/eip-7668.md

```diff
@@ -12,23 +12,23 @@ created: 2024-03-31
 
 ## Abstract
 
-Require the bloom filters in an execution block, including at the top level and in the receipt object, to be empty.
+Require the bloom filters in an execution block, including at the top level and in the receipt object, to be hardcoded to `0xffff...ff`.
 
 ## Motivation
 
 Logs were originally introduced to give applications a way to record information about onchain events, which decentralized applications (dapps) would be able to easily query. Using bloom filters, dapps would be able to quickly go through the history, identify the few blocks that contained logs relative to their application, and then quickly identify which individual transactions have the logs that they need.
 
-In practice, this mechanism is far too slow. Almost all dapps that access history end up doing so not through RPC calls to an Ethereum node (even a remote-hosted one), but through centralized extra-protocol services.
+In practice, this mechanism is far too slow. Almost all dapps that access history end up doing so not through RPC calls to an Ethereum node (even a remote-hosted one), but through centralized extra-protocol services.  Not too mention, in a chain with relatively full blocks, the false positive rate in bloom filters has increased to the point of making bloom filters almost useless.
 
-This EIP proposes recognizing that reality, and removing bloom filters from the protocol. Applications that need history querying would be encouraged to develop and use decentralized protocols that create provable log indexes using eg. ZK-SNARKs and incrementally-verifiable computation.
+This EIP proposes recognizing that reality, and removing bloom filters from the protocol since it places an unnecessary compute burden on execution clients computing what is for all prati. Applications that need history querying would be encouraged to develop and use decentralized protocols that create provable log indexes using eg. ZK-SNARKs and incrementally-verifiable computation.
 
 ## Specification
 
-The logs bloom of an execution block is now required to be empty (ie. 0 bytes long). The logs bloom of a transaction receipt is now required to be empty (ie. 0 bytes long).
+The `logsbloom` field of the execution block header and the `logsbloom` field of the transaction receipt is now required to be `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` (ie. 32 bytes filled with `ff`).
```

Comment:
```suggestion
The `logsBloom` field of the execution block header and the `logsBloom` field of the transaction receipt is now required to be `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` (ie. 256 bytes filled with `ff`).
```

Reply:
I guess cut it off (the 0xffff..ff), because this is wayyy too long.

Also, I think that the original (empty bytes it instead of writing the 1s) is better for storage and performance reasons. I know we wanted to write the all-bits-1 version because then the bloom filter would be positive for any query, but I think this can be handled at the application side (RPC). We could even wrap the RPC such that it returns the 0xffff..ffff as `logsBloom` for instance, or just even better really deprecate `logsBloom` and thus writing empty bytes.

We could also consider completely removing the `logsBloom` field from the receipts. This would save some extra bytes when hashing the receipt for the `receiptsTrie`.

Reply:
Are you saying replace the current field with a static byte array of 256 bytes of 0 or just an empty array?   I'm not confident on the implications for the RLP encoding of a tx receipt or block header.  I guess maybe nothing if we say as of the fork block, the logsBloom field becomes a `0` or `[]`, right?

Reply:
If we input the empty bytes for the `logsBloom` starting at fork block then the RLP structure will not change (besides that the field is now not the constant 256 bytes long, but now 0. This is not a problem, we have variable-length fields like `extraData` which can be something between 0-32 bytes). If we would input it as empty list then this would (from my perspective) change the structure somewhat. I'd say empty bytes is the way to go. So empty bytes is not the empty list `[]`, it is a field of length 0. (This is possible, txs with a `to` field of 0 bytes are CREATE txs)